lib/server/middleware.js

```diff
@@ -0,0 +1,507 @@
+ *
+ * @param {RegExp} validContentTypeRe - Regex to validate content-type format
+ * @return {Function} Restify pre-middleware function
+ */
+function createCleanupContentType(validContentTypeRe) {
+    return function cleanupContentType(req, res, next) {
+        var ct = req.headers['content-type'];
+        /*
+         * content-type must have a type, '/' and sub-type
+         */
+        if (ct && !validContentTypeRe.test(ct)) {
+            req.log.debug('receieved a malformed content-type: %s', ct);
+            req.headers['content-type'] = mime.lookup(ct);
+        }
+
+        next();
+    };
+}
+
+
+/**
+ * Pre-middleware to preserve raw request body for S3 signature verification.
+ * Captures and buffers request body for POST/PUT operations that require
+ * signature verification (multipart uploads, bulk deletes, STS, CORS config).
+ *
+ * @param {Object} req - Restify request object
+ * @param {Object} res - Restify response object
+ * @param {Function} next - Restify next callback
+ */
+function preserveRawBodyPreMiddleware(req, res, next) {
+    // Check if this is a Complete Multipart Upload request
+    var isCompleteMultipartUpload = false;
+    if (req.method === 'POST' && req.url) {
+        // Simple check for uploadId query parameter
+        // (Complete Multipart Upload)
+        isCompleteMultipartUpload = req.url.includes('uploadId=') &&
+                                   !req.url.includes('uploads');
+    }
+
+    // Check if this is a CORS configuration request
+    var isCorsRequest = req.method === 'PUT' && req.url &&
+                       req.url.includes('?cors');
+
+    // Preserve raw body for POST requests that might be S3 bulk operations
+    // Include: XML content-type requests, Complete Multipart Upload
+    // requests,
+    // bulk delete requests, OR STS requests
+    // Also preserve for PUT CORS configuration requests
+    var contentType = req.headers['content-type'] || '';
+    var isBulkDeleteRequest = req.url && req.url.includes('delete');
+    var isXmlRequest = contentType === 'application/xml' ||
+                      contentType.startsWith('application/x-amz-json') ||
+                      contentType.startsWith('text/xml');
+
+    // Check if this is an STS request (POST to / with Action parameter)
+    var isStsRequest = req.method === 'POST' &&
+                      req.url === '/' &&
+                      req.headers.authorization &&
+                      req.headers.authorization.toLowerCase().
+        indexOf(constants.AWS_AUTH.SCHEME_SIGV4) !== -1;
+
+    if (((req.method === 'POST' &&
+          req.headers.authorization &&
+          (isXmlRequest || isCompleteMultipartUpload ||
+           isBulkDeleteRequest || isStsRequest)) ||
+         (req.method === 'PUT' && isCorsRequest)) &&
+        req.headers['content-length'] &&
+        parseInt(req.headers['content-length'], 10) > 0) {
+
+        req.log.debug({
+            method: req.method,
```

Comment:
Just a suggestion if you weren't already aware. If you pass the whole `req` object to bunyan it will extract the [method, url, and headers](https://github.com/trentm/node-bunyan?tab=readme-ov-file#standard-serializers) for you.

If this is intentional and/or you were already aware feel free to disregard/resolve.  You might have good reasons for passing in explicit params too I just noticed the debug calls has a lot of properties. A custom serializer for common fields might be worth considering too.